docs/next/guides/upgrade-and-migration/migrating-from-11.1-to-12.0.md

```diff
@@ -12,3 +12,116 @@ pageClass: migration-guide
+
+- Instead of passing the [`data`](@/api/options.md#data) option to the [`updateSettings()`](@/api/core.md#updatesettings) method, run [`loadData()`](@/api/core.md#loadData) and provide the new dataset as its argument:
+```js
+// Handsontable 12.0: this doesn't reset your cell meta and index mapper configuration
+hotInstance.updateSettings({
+  data: newDataset
+});
+
+// Handsontable 12.0: this does reset your cell meta and index mapper configuration
+hotInstance.loadData(newDataset);
+```
+
+#### Framework wrappers
+
+Updating your [`data`](@/api/options.md#data) through a component property no longer resets your [index mappings](@/api/indexmapper.md) and [configuration options](@/guides/getting-started/setting-options.md) (`CellMeta`).
+
+To replace [`data`](@/api/options.md#data) and reset the states, call the [`loadData()`](@/api/core.md#loadData) method (the same way as above), referencing the Handsontable instance from the component.
+
+Read more on referencing the Handsontable instance:
+- [Referencing the Handsontable instance in Angular](@/guides/integrate-with-angular/angular-hot-reference.md)
+- [Referencing the Handsontable instance in React](@/guides/integrate-with-react/react-hot-reference.md)
+- [Referencing the Handsontable instance in Vue 2](@/guides/integrate-with-vue/vue-hot-reference.md)
+- [Referencing the Handsontable instance in Vue 3](@/guides/integrate-with-vue3/vue3-hot-reference.md)
+
+## Step 2: Adjust to the `updatePlugin()` changes
+
+Handsontable [12.0.0](https://github.com/handsontable/handsontable/releases/tag/12.0.0) changes the way the [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method reacts to [`updateSettings()`](@/api/core.md#updatesettings) calls.
+
+#### Before
+
+Every [`updateSettings()`](@/api/core.md#updatesettings) call (even with an empty object passed as new settings) triggered 
+the [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method for each enabled plugin.
+
+As a result, whenever you called [`updateSettings()`](@/api/core.md#updatesettings), all enabled plugins got updated.
+
+#### After
+
+A plugin gets updated only if you update settings related to that particular plugin.
+
+Only the following plugins still get updated on every [`updateSettings()`](@/api/core.md#updatesettings) call:
+  - [`AutoColumnSize`](@/api/autocolumnsize.md)
+  - [`AutoRowSize`](@/api/autorowsize.md)
+  - `TouchScroll`
+  - [`UndoRedo`](@/api/undoredo.md)
+
+#### Migrating to Handsontable 12.0
+
+If you want a plugin (e.g. [`ContextMenu`](@/api/contextmenu.md)) to still get updated every time you call [`updateSettings()`](@/api/core.md#updatesettings), use the [`afterUpdateSettings`](@/api/hooks.md#afterupdatesettings) hook to call your plugin's [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method:
```

Comment:
@jansiegel Why would I want `ContextMenu` to update every time I call `updateSettings`? Everyone including me will ask how do they know if they are affected by this step of the migration guide. 

Shouldn't the migration advice here be centered on how are **custom** plugins affected?

Reply:
@warpech I think `ContextMenu` is used just as an example of a plugin. A custom plugin example would probably make more sense, true. 
@kirszenbaum

Reply:
Updated: [e92ee45](https://github.com/handsontable/handsontable/pull/9271/commits/e92ee45ad605d1cd9c0f378e571cd27ff188a8a4)